docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,237 @@ The remaining points can happen any time after 1) is complete although note that
+   [cylc-flow #3819](https://github.com/cylc/cylc-flow/issues/3819)
+   and [rose #2412](https://github.com/metomi/rose/pull/2412).
 
 6. Utility to remove installed locations (on all platforms) on demand
    (equivalent to `rose suite-clean`).
 
-7. On starting up a new run of the suite assign a new run name:
+7. Use a different run directory for each installation of a suite:
    ```
-   mi-aa001/
+   u-ab123/
      run1
      run2
    ```
-   * This is new functionality but Cylc 8 is a good time to introduce it.
+   * This is big change in working practice but Cylc 8 is a good time to
+     introduce it.
    * Use a symbolic link to identify the latest run which is the default name
      when performing operations on a suite?
 
-8. Consider the behaviour created by the `CYLC_VERSION` and `ROSE_VERSION`
-   variables in `rose-suite.conf`. These allow rose suite-run to:
-   * Change the version of Cylc and check the version of Rose used to run the
-     suite.
-   * Prevent suites accidentally upgrading or downgrading versions on restart.
-
-9. Consider `rose prune` functionality.
+8. Consider `rose prune` functionality.
    * Need to make sure it will continue to work correctly.
-   * Need Cylc equivalent but this should be done as part of
+   * Longer term we should build housekeeping into Cylc, see
      [cylc-flow #1159](https://github.com/cylc/cylc-flow/issues/1159).
 
-### CLI
+9. Record version control information when you install a suite, see
+   [cylc-flow #3849](https://github.com/cylc/cylc-flow/issues/3849).
+   * `rose suite-run` current supports this for Subversion only
+     (`svn info` + `svn status` + `svn diff`).
+   * Extend this to support equivalent information for git (and preferably make
+     it easy to plugin support for other systems as well).
+
+### CLI changes
+
+`cylc install` - new command to install workflows.
+* By default, `cylc install` will install the workflow found in `$PWD` into
+  `~/cylc-run/$(basename $PWD)/runN` (where `runN` = `run1`, `run2` ...).
+* If `--run-name` is not specified and `run1` already exists it will install
+  into `run2` (and so on).
+  Create a symlink `runN` pointing at the latest run.
+* `--run-name=my-run` implies install into `~/cylc-run/$(basename $PWD)/my-run`.
+  If the target directory already exists then fail.
+* `--no-run-name` implies install into `~/cylc-run/$(basename $PWD)`.
+* `--flow-name=my-flow` implies install into `~/cylc-run/my-flow/runN`.
+* `--directory=/path/to/flow` (`-C ...`) implies install the workflow found in
+  `/path/to/flow` (rather than `$PWD`).
+* Installation will involved copying over the files found in the source
+  directory.
+  * This will exclude any `.git`, `.svn`, `log`, share` or `work` directories.
+    Everything else will be copied.
+    * If needed we could make this configurable via a `.cylcignore` file (future
+      enhancement). Note that `log`, share` & `work` would always be excluded.
```

Comment:
(Why would there be `log`, `share`, and `work` in the source directory?)

Reply:
(Maybe installation should abort if those are found, as it would suggest a run directory not a source directory)

Reply:
> (Why would there be log, share, and work in the source directory?)

There shouldn't be! This change makes it an error to have any of these directories in the src dir to avoid the nasty results of rsync'ing one over the other.

Reply:
> Note that `log`, share` & `work` would always be excluded.

This says those directories will be excluded, but it doesn't make it an error for them to exist.  IMO it should be and error. (Or have I misunderstood?)

Reply:
Ah, sorry, I'm getting conversations mixed up with the proposal.

Yes, this would silently ignore these dirs. I agree this should be an error or at the every least a warning.

Reply:
Proposal updated to make this an error